### PR TITLE
CLS2-675: add similar task

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
## Description of change

This PR allows the users to create a task starting from a pre existing one. The new task should not have `title` `description` and `id` so that a new one gets created.

[Jira ticket](https://uktrade.atlassian.net/browse/CLS2-675)

## Test instructions

- Create a new task
- In the task details page click on create similar task
- Fill in the title field
- The task list page should now show the new task alongside the existing one

### After
Screenshot 2024-01-11 at 15 14 00
Screenshot 2024-01-11 at 15 14 11
Screenshot 2024-01-11 at 15 14 22
Screenshot 2024-01-11 at 15 14 29

_Add a screenshot_

## Checklist

- [x] Has the branch been rebased to main?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/modules/Tasks/TaskForm/state.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const TASK_SAVE_TASK_DETAILS = 'TASK_SAVE_TASK_DETAILS'
 
 export const getGenericBreadcumbs = (task) => {
   const defaultBreadcrumbs = [{ link: urls.dashboard.myTasks(), text: 'Home' }]
-  if (task) {
+  if (task && task.title?.length > 0) {
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskForm/state.js:18]: Can just be `if (task && task.title) {`

File: src/client/modules/Tasks/TaskForm/TaskFormAdd.jsx [hunk 2]
```diff
@@ -97,6 +99,16 @@ const getTaskProps = (investmentProjectId, interactionId) => {
...
+      id: TASK_ID,
+      startOnRender: {
+        payload: copyTaskId,
+        onSuccessDispatch: COPY_TASK_DETAILS_LOADED,
+      },
+    }
+  }
 }
 
 const TaskFormAdd = ({
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskForm/TaskFormAdd.jsx:114]: This form is giving a null for the title, there is the `getTitle` function in this file that will fix it with some changes

image

File: test/functional/cypress/specs/tasks/create-similar-task-spec.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import { fill } from '../../support/form-fillers'
+import {
+  assertPayload,
+  assertFlashMessage,
+} from '../../../cypress/support/assertions'
+import { DATE_LONG_FORMAT_3 } from '../../../../../src/common/constants'
+import { format } from '../../../../../src/client/utils/date'
+
+describe('Copy task from generic task task', () => {
+  const genericTask = taskFaker({ archived: false })
```
Comment (chopkinsmade) [inline test/functional/cypress/specs/tasks/create-similar-task-spec.js:17]: `archived` is false by default, no need to provide this here or in the usages below

File: src/client/modules/Tasks/TaskDetails/TaskButtons.jsx [hunk 4]
```diff
@@ -43,15 +43,26 @@ export const TaskButtons = ({ task, returnUrl }) => (
...
+            <Button
+              buttonColour={GREY_3}
+              buttonTextColour={TEXT_COLOUR}
+              as={Link}
+              href={urls.tasks.edit(task.id)}
+              data-test="edit-form-button"
+            >
+              Edit
+            </Button>
+            <Button
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskDetails/TaskButtons.jsx:56]: This isn't part of the ticket, but should this button be here even for completed tasks? I'd check with isabel

Comment (samuele-mattiuzzo) [reply src/client/modules/Tasks/TaskDetails/TaskButtons.jsx:56]: I'll ask Isabel, I can still pick it up next week if it needs changing

Comment (samuele-mattiuzzo) [reply src/client/modules/Tasks/TaskDetails/TaskButtons.jsx:56]: @chopkinsmade I've left a comment in the jira issue and tagged both Marijn and Isabel

**Review by chopkinsmade — APPROVED** [review]

**Review by ClaudiaGC1339 — APPROVED** [review]

**Pull request merged**